tests/func/test_get.py

```diff
@@ -6,178 +6,185 @@
+    external_data.write_text("ext_data")
+
+    with monkeypatch.context() as m:
+        m.chdir(fspath(erepo_dir))
+        erepo_dir.dvc.add(fspath(external_data))
+        erepo_dir.scm_add(["ext_data.dvc"], commit="add external data")
+
+    Repo.get(fspath(erepo_dir), fspath(external_data), "ext_data_imported")
+    assert (tmp_dir / "ext_data_imported").is_file()
+    assert (tmp_dir / "ext_data_imported").read_text() == "ext_data"
+
+
+def test_non_cached_output(tmp_dir, erepo_dir):
+    src = "non_cached_file"
+    dst = src + "_imported"
+
+    erepo_dir.dvc.run(
+        outs_no_cache=[src],
+        cmd="echo hello > non_cached_file",
+        cwd=fspath(erepo_dir),
     )
-    erepo.dvc.scm.add(["non_cached_file", "non_cached_file.dvc"])
-    erepo.dvc.scm.commit("add non-cached output")
-    os.chdir(fspath_py35(tmp_path))
-    Repo.get(erepo.root_dir, "non_cached_file")
+    erepo_dir.scm.add([src, src + ".dvc"])
+    erepo_dir.scm.commit("add non-cached output")
 
-    src = os.path.join(erepo.root_dir, "non_cached_file")
-    assert os.path.isfile("non_cached_file")
-    assert filecmp.cmp(src, "non_cached_file", shallow=False)
+    Repo.get(fspath(erepo_dir), src, dst)
+
+    assert (tmp_dir / dst).is_file()
+    # NOTE: using strip() to account for `echo` differences on win and *nix
+    assert (tmp_dir / dst).read_text().strip() == "hello"
 
 
 # https://github.com/iterative/dvc/pull/2837#discussion_r352123053
-def test_absolute_file_outside_repo(erepo):
+def test_absolute_file_outside_repo(tmp_dir, erepo_dir):
     with pytest.raises(PathMissingError):
-        Repo.get(erepo.root_dir, "/root/")
+        Repo.get(fspath(erepo_dir), "/root/")
 
 
-def test_unknown_path(erepo):
+def test_unknown_path(tmp_dir, erepo_dir):
     with pytest.raises(PathMissingError):
-        Repo.get(erepo.root_dir, "a_non_existing_file")
+        Repo.get(fspath(erepo_dir), "a_non_existing_file")
 
 
 @pytest.mark.parametrize("dname", [".", "dir", "dir/subdir"])
-def test_get_to_dir(dname, erepo):
-    src = erepo.FOO
+def test_get_to_dir(tmp_dir, erepo_dir, monkeypatch, dname):
+    with monkeypatch.context() as m:
+        m.chdir(fspath(erepo_dir))
+        erepo_dir.dvc_gen("foo", "foo", commit="create foo")
 
     makedirs(dname, exist_ok=True)
 
-    Repo.get(erepo.root_dir, src, dname)
+    Repo.get(fspath(erepo_dir), "foo", dname)
 
-    dst = os.path.join(dname, os.path.basename(src))
+    assert (tmp_dir / dname).is_dir()
+    assert (tmp_dir / dname / "foo").read_text() == "foo"
 
-    assert os.path.isdir(dname)
-    assert filecmp.cmp(erepo.FOO, dst, shallow=False)
 
+def test_get_from_non_dvc_master(
+    tmp_dir, erepo_dir, tmp_path, monkeypatch, caplog
+):
+    with monkeypatch.context() as m:
+        m.chdir(fspath(erepo_dir))
+        erepo_dir.scm.checkout("new_branch", create_new=True)
+        erepo_dir.scm_gen({"foo": "foo"}, commit="create foo")
+        erepo_dir.scm.checkout("master")
```

Comment:
You slightly changed the logic here. Previously foo existed in both branches, now only in new. At least this is what the code says.

Things made worse by the fact that `erepo_dir` has repo template. So I suggest:
- create a file and commit to master branch, use other name than foo or bar
- create a new branch, no need to even checkout to it
- remove `.dvc` in master
- try get from master, check that this doesn't work. This would be a sanity check for those numerous prep steps
- get from rev

Reply:
Got it. I assumed that we don't have templates anymore. Will adjust a bit differently and will look throught other tests. I should've started participating in conversion earlier 🙂